### ISSUE TITLE
Bitmessage v0.5.6 doesn't run on Windows 10 Pro

### ISSUE THREAD

**Issue by HostFat** [mailchuck/PyBitmessage#152]
From debug.log

2015-12-21 17:50:30,269 - INFO - Loaded C PoW DLL (stdcall) bitmsghash32.dll
2015-12-21 17:50:30,269 - DEBUG - C PoW start
2015-12-21 17:50:30,269 - ERROR - C PoW test fail.
Traceback (most recent call last):
  File "C:\Dokumente und Einstellungen\Besitzer\src\PyInstaller\bitmessagemain\build\bitmessagemain\out00-PYZ.pyz\proofofwork", line 171, in <module>
  File "C:\Dokumente und Einstellungen\Besitzer\src\PyInstaller\bitmessagemain\build\bitmessagemain\out00-PYZ.pyz\proofofwork", line 86, in _doCPoW
ValueError: Procedure probably called with too many arguments (12 bytes in excess)
2015-12-21 17:50:30,269 - INFO - Loaded C PoW DLL (cdecl) bitmsghash32.dll
2015-12-21 17:50:30,269 - DEBUG - C PoW start
2015-12-21 17:50:30,270 - DEBUG - C PoW done
2015-12-21 17:50:30,270 - INFO - Successfully tested C PoW DLL (cdecl) bitmsghash32.dll
2015-12-21 17:50:31,052 - CRITICAL - PyBitmessage requires PyQt unless you want to run it as a daemon and interact with it using the API. You can download it from http://www.riverbankcomputing.com/software/pyqt/download or by searching Google for 'PyQt Download' (without quotes).
Traceback (most recent call last):
  File "C:\Dokumente und Einstellungen\Besitzer\src\PyInstaller\bitmessagemain\build\bitmessagemain\out00-PYZ.pyz\bitmessageqt", line 14, in <module>
  File "C:\Dokumente und Einstellungen\Besitzer\src\PyInstaller\PyInstaller\loader\pyi_importers.py", line 409, in load_module
ImportError: DLL load failed: A dynamic link library (DLL) initialization routine failed.

**Comment by PeterSurda** [comment]
That's odd, it works on my Win10.

**Comment by PeterSurda** [comment]
The first error is not a problem, it just tries to find out whether the PoW DLL was compiled with MinGW or MSVS. The problem is that it can't find pyqt for some reason.

**Comment by PeterSurda** [comment]
It works fine on my wife's 8.1 and that computer has no development libraries. Do you have QT, python, or pyqt installed?

**Comment by HostFat** [comment]
No, I think not ...

If you give me a version with a more debug info on the log, I'll run it.

**Comment by PeterSurda** [comment]
This is the max logging verbosity that you can have. Try to install https://www.microsoft.com/en-us/download/details.aspx?id=29 (Microsoft Visual C++ 2008 Redistributable Package (x86)) and let me know if that helps.

**Comment by PeterSurda** [comment]
To myself: this is a possible fix for missing MSVC++: http://www.py2exe.org/index.cgi/Tutorial#Step52

**Comment by HostFat** [comment]
I had it already installed, it asked to uninstall or "repair" the installation.

I clicked "repair".

It still doesn't work.

**Comment by PeterSurda** [comment]
Try the Dependency Walker http://www.dependencywalker.com/
and tell me what's missing.

**Comment by HostFat** [comment]
This include my Bitmessage.dwi file, is it what you wanted?

[Bitmessage.zip](https://github.com/mailchuck/PyBitmessage/files/68806/Bitmessage.zip)

**Comment by PeterSurda** [comment]
This is what I'd like you to do once inside dependency walker. First, get the x86 version of DW (unless you already did). Then, open the bitmessage-0.5.6.exe with it. Then, click Profile -> Start profiling. Disable all checkboxes and click ok. It should crash after a while and explain where. I'm not familiar with DW but I think this is how it should work. It doesn't crash on my machine so I don't know what exactly will happen.

**Comment by HostFat** [comment]
Here again the new file with the DW x86 version after the profiling.

[Bitmessage.zip](https://github.com/mailchuck/PyBitmessage/files/68825/Bitmessage.zip)

**Comment by PeterSurda** [comment]
Well unfortunately I don't know. I'm not a Windows developer. I'll try to setup a bounty to solve this one.

**Comment by Fabio3rs** [comment]
I installed PyQt for Python 3 by mistake and it gave the same error, then I installed the PyQt for Python 2.7 and the program works.

HostFat, make sure you have PyQt for Python 2.7, not for Python 3

https://www.riverbankcomputing.com/software/pyqt/download
Download either PyQt4-4.11.4-gpl-Py2.7-Qt4.8.7-x64.exe   or PyQt4-4.11.4-gpl-Py2.7-Qt4.8.7-x32.exe, notice how there's a -Py2.7 appeneded to the filename

Before I install PyQt for Python 2.7:

```
C:\Users\Fabio\Downloads\PyBitmessage-master\PyBitmessage-master\src>bitmessagemain.py
Loading existing config files from C:\Users\Fabio\AppData\Roaming\PyBitmessage\
2015-12-22 00:35:56,865 - ERROR - C PoW test fail.
Traceback (most recent call last):
  File "C:\Users\Fabio\Downloads\PyBitmessage-master\PyBitmessage-master\src\proofofwork.py", line 167, in <module>
    bso = ctypes.WinDLL(os.path.join(codePath(), "bitmsghash", bitmsglib))
  File "C:\Python27\lib\ctypes\__init__.py", line 365, in init
    self._handle = _dlopen(self._name, mode)
WindowsError: [Error 126] NÒo foi possÝvel encontrar o m¾dulo especificado
2015-12-22 00:35:56,881 - ERROR - C PoW test fail.
Traceback (most recent call last):
  File "C:\Users\Fabio\Downloads\PyBitmessage-master\PyBitmessage-master\src\proofofwork.py", line 177, in <module>
    bso = ctypes.CDLL(os.path.join(codePath(), "bitmsghash", bitmsglib))
  File "C:\Python27\lib\ctypes\__init__.py", line 365, in init
    self._handle = _dlopen(self._name, mode)
WindowsError: [Error 126] NÒo foi possÝvel encontrar o m¾dulo especificado
2015-12-22 00:35:56,959 - CRITICAL - PyBitmessage requires PyQt unless you want to run it as a daemon and interact with it using the API. You can download it from http://www.riverbankcomputing.com/software/pyqt/download or by searching Google for 'PyQt Download' (without quotes).
Traceback (most recent call last):
  File "C:\Users\Fabio\Downloads\PyBitmessage-master\PyBitmessage-master\src\bitmessageqt\__init__.py", line 11, in <module>
    from PyQt4 import QtCore, QtGui
ImportError: No module named PyQt4
2015-12-22 00:35:56,960 - CRITICAL -   File "C:\Users\Fabio\Downloads\PyBitmessage-master\PyBitmessage-master\src\bitmessagemain.py", line 269, in <module>
    mainprogram.start()
  File "C:\Users\Fabio\Downloads\PyBitmessage-master\PyBitmessage-master\src\bitmessagemain.py", line 161, in start
    thisapp = singleton.singleinstance("", daemon)
  File "C:\Users\Fabio\Downloads\PyBitmessage-master\PyBitmessage-master\src\singleton.py", line 24, in init
    import bitmessageqt
  File "C:\Users\Fabio\Downloads\PyBitmessage-master\PyBitmessage-master\src\bitmessageqt\__init__.py", line 19, in <module>
    sys.exit()

2015-12-22 00:35:56,961 - CRITICAL - <type 'exceptions.NameError'>: name 'sys' is not defined
```

After the installation of PyQt for Python 2.7 (PyQt4-4.11.4-gpl-Py2.7-Qt4.8.7-x64.exe):
![img](https://cloud.githubusercontent.com/assets/9311162/11946905/cf5920fe-a846-11e5-8c3a-22920a2c5118.png)

**Comment by HostFat** [comment]
After the installation of "PyQt4-4.11.4-gpl-Py2.7-Qt4.8.7-x32.exe" it works. (it doesn't work instead with the x64 version)

Anyway, I think that the common user should be not forced to install these kind of package/libraries to use the software.

They should be included with the package.

**Comment by Fabio3rs** [comment]
Perhaps Qt dlls should be distributed with the frozen executable?

**Comment by PeterSurda** [comment]
I did the tests on XP without development libraries (which includes PyQT) and 8.1 without development libraries and it works. The problem must be elsewhere.

**Comment by PeterSurda** [comment]
Maybe it's about the priority of the library loading then.

**Comment by PeterSurda** [comment]
I made some progress. If you have a wrong version of PyQT installed, it seems to take precedence over the one that's in the exe. When I installed PyQt4-4.11.4-gpl-Py3.4-Qt4.8.7-x32.exe on my test machine, it started complaining that MSVCR100.DLL is missing (but it still launches). So the solution is to tell the system that the bundled PyQT should take precedence over the one that's on the system. However, I don't know how.

**Comment by HostFat** [comment]
This is the first result that I've found on Google, I hope that it will be useful in some way :)

http://arstechnica.com/information-technology/2009/03/how-to-deploying-pyqt-applications-on-windows-and-mac-os-x/

**Comment by PeterSurda** [comment]
I received reports that the same problem occurs on Windows XP SP3. However, my test machine with Windows XP SP3 works fine without a PyQT installed.

**Comment by PeterSurda** [comment]
Here is the spec file that I use with pyinstaller: https://mega.nz/#!8AtkTCBB!ahUZkf4089gXmO7Xm_J7rFAwf4fDkroZ0rAKi-bLYls

**Comment by feroldi** [comment]
Are you able to compile PyQt statically?

Anyway, you should be able to hint PyQt a path to look for dlls.

**Comment by PeterSurda** [comment]
I got a report over bitmessage that the reason for this is that it finds old libeay32.dll in a PATH (say c:\windows) and this is loaded first, and then it can't load QtNetwork4.dll because that is linked against a different library version. The workaround, for the time being, apart from installing the correct pyqt locally, is to delete old libeay32.dll.

**Comment by thelink2012** [comment]
Interesting. Would you, as kinda of mentioned before by Fabio3rs, perhaps redistribute libeay32.dll with the frozen exe, then it'll take priority over the one in PATH?

**Comment by PeterSurda** [comment]
The libeay32.dll is in the exe (I was actually told it isn't the one from OpenSSL 1.0.2e as I intended but 1.0.1h, but that's a different issue), just the ones in PATH seem to take precedence for some reason. If it's not the right version, then it's not binary compatible with the bundled PyQT and QtNetwork fails to load (at least that's how I understand it).

**Comment by feroldi** [comment]
If it's a problem of order, try inserting the pyqt path into `sys.path`, so that such one should be checked first.

```
if sys.platform.startswith('win32') or sys.platfrom.startswith('cygwin'):
    import sys
    sys.path.insert(1,'/path/to/pyqt_directory')
```

**Comment by PeterSurda** [comment]
I can now confirm that the bug description I got over BM was correct: incompatible libeay32.dll in PATH. Also the recommended fix, rebuilding with PyInstaller 3.0 instead of 2.1, solves the problem. I should be able to release 0.5.7 in the next day or two, which will have this fixed.
